src/components/steppers/FooterStepper.vue

```diff
@@ -0,0 +1,141 @@
+<script lang="ts" setup>
+import RuiIcon from '@/components/icons/Icon.vue';
+import RuiButton from '@/components/buttons/Button.vue';
+import RuiProgress from '@/components/progress/Progress.vue';
+
+const props = withDefaults(
+  defineProps<{
+    pages: number;
+    modelValue: number;
+    type?: 'numeric' | 'bullet' | 'progress' | 'pill';
+  }>(),
+  {
+    type: 'numeric',
+  }
+);
+
+const emit = defineEmits<{
+  (event: 'update:modelValue', modelValue: number): void;
+}>();
+
+const css = useCssModule();
+
+const { modelValue, pages } = toRefs(props);
+
+const onChange = (delta: number) => {
+  emit('update:modelValue', get(modelValue) + delta);
+};
+
+const onPrev = () => {
+  const value = get(modelValue);
+  if (value) {
+    onChange(-1);
+  }
+};
+const onNext = () => {
+  const value = get(modelValue);
+  if (value < get(pages)) {
+    onChange(1);
+  }
+};
+
+const onClick = (index: number) => {
+  onChange(index - get(modelValue));
+};
+</script>
+
+<template>
+  <div :class="[css['footer-stepper'], css[type ?? '']]">
+    <template v-if="type === 'pill'">
+      <div :class="css.pills">
+        <span
+          v-for="i in pages"
+          :key="i"
+          :class="[css.pill, { [css.active]: modelValue === i }]"
+          @click="onClick(i)"
+        />
+      </div>
+    </template>
+    <template v-else>
+      <rui-button
+        :disabled="modelValue <= 1"
+        color="primary"
+        tile
+        variant="text"
+        @click="onPrev()"
+      >
+        <template #prepend>
+          <rui-icon class="-ml-[.8rem]" name="arrow-left-s-line" />
+        </template>
```

Comment:
```
        <template #prepend>
          <rui-icon size="18" name="arrow-left-s-line" />
        </template>
```

Reply:
let's reduce the size right, and remove negative margin? ok